FanGraphSpider.py

```diff
@@ -1,174 +0,0 @@
-import scrapy
-import tweepy
-import datetime
-from scrapy_splash import SplashRequest
-import csv
-from scrapy.crawler import CrawlerProcess
-
-
-tkey = 'Redacted'
-tsecret = 'Redacted'
-atoke = 'Redacted'
-asecret = 'Redacted'
-
-teamdict = {
-    'Atlanta Braves': 'Braves',
-    'Miami Marlins': 'Marlins',
-    'New York Mets': 'Mets',
-    'Philadelphia Phillies': 'Phillies',
-    'Washington Nationals': 'Nationals',
-    'Chicago Cubs': 'Cubs',
-    'Cincinnati Reds': 'Reds',
-    'Milwaukee Brewers': 'Brewers',
-    'Pittsburgh Pirates': 'Pirates',
-    'St. Louis Cardinals': 'Cardinals',
-    'Arizona Diamondbacks': 'Diamondbacks',
-    'Colorado Rockies': 'Rockies',
-    'Los Angeles Dodgers': 'Dodgers',
-    'San Diego Padres': 'Padres',
-    'San Francisco Giants': 'Giants',
-    'Baltimore Orioles': 'Orioles',
-    'Boston Red Sox': 'Red%20Sox',
-    'New York Yankees': 'Yankees',
-    'Tampa Bay Rays': 'Rays',
-    'Toronto Blue Jays': 'Blue%20Jays',
-    'Chicago White Sox': 'White%20Sox',
-    'Cleveland Indians': 'Indians',
-    'Detroit Tigers': 'Tigers',
-    'Kansas City Royals': 'Royals',
-    'Minnesota Twins': 'Twins',
-    'Houston Astros': 'Astros',
-    'Los Angeles Angels': 'Angels',
-    'Oakland Athletics': 'Athletics',
-    'Seattle Mariners': 'Mariners',
-    'Texas Rangers': 'Rangers',
-}
-
-#Builds URLs to be scraped from team name and date. Team Name Gathered from MLB SPider.
-starturls = []
-dt = datetime.date.today().strftime('%Y-%m-%d')
-with open('schedule.csv', 'r') as input:
-    reader = csv.reader(input)
-    for item in reader:
-        # Only adds item to url if time is later than scheduled game time
-        if item[0] in teamdict.keys() and item[1] < datetime.datetime.now().strftime('%H:%M'):
-            starturls.append('https://www.fangraphs.com/livewins.aspx?date=%s&team=%s&dh=0&season=%s' % (dt, teamdict[item[0]], dt[:4]))
-
-
-class FanGraphSpider(scrapy.Spider):
-    name = 'FanGraphSpider'
-    start_urls = starturls
-    SPLASH_ARGS = {
-        'wait': 20.0,
-        'images': 0,
-        'render_all': 1,
-        'timeout': 40.0,
-        'resource_timeout': 40.0}
-
-    def parse(self, response):
-        # Checks for Home Runs and stolen bases in play log and, if found, sends player stat page to parser.
-        tweeted = False
-        for item in response.xpath('//div[@id="LiveGame1_dgPlay"]/table/tbody/tr'):
-            if 'homered' in item.xpath('td[6]/text()').extract_first():
-                # Checks to see if play has already been sent to parser. If not, adds it to list to avoid duplicates.
-                plname = item.xpath('td[2]/text()').extract_first()
-                with open('dupcatcher.csv', 'r') as input:
-                    reader = csv.reader(input)
-                    for line in reader:
-                        if [item.xpath('td[6]/text()').extract_first()] == line:
-                            tweeted = True
-                if not tweeted:

```

Comment:
Oh another note here, I always get really nervous and suspicious when the tab depth gets this far in, and that's usually a good sign of code starting to get too complicated and tough to follow. It's a good idea to pull out some utility methods to prevent this, here's a possible example:
```
with open('dupcatcher.csv', 'r') as input:
    reader = csv.reader(input)
    for line in reader:
        if [item.xpath('td[6]/text()').extract_first()] == line:
            tweeted = True
if not tweeted:
```
becomes
```
def is_tweeted():
    with open('dupcatcher.csv', 'r') as input:
        reader = csv.reader(input)
        for line in reader:
            if [item.xpath('td[6]/text()').extract_first()] == line:
                return True
    return False
...
if not is_tweeted():
```